### DISCUSSION TITLE
Get source IP when accessing an application exposed through Traefik Ingress behind a Haproxy LB

### DISCUSSION THREAD

**Discussion by toni123321** [k3s-io/k3s#10409] [Q&A]
I have a k3s cluster with Traefik and I am exposing my applications using Traefik ingress controllers. 
I have tried to place Haproxy or Azure Managed LB in front of my k3s control plane node/server node but then I got not my source IP but some private IPs like the private IP of the Traefik pod. Why is that? 

I have already tried adding externalTrafficPolicy: Local in HelmChartConfig as well as other properties from https://github.com/k3s-io/k3s/discussions/2997 but without success.

**Comment by brandond** [comment]
Are you still using ServiceLB? What ports are you targeting with the external LB? If you're not using a LoadBalancer controller that knows what to do with externalTrafficPolicy on the service, then you should manually target the service's NodePort, instead of targeting the LB port. There are implementation details that you're likely missing by configuring your LB by hand, and expecting it to just work by pointing at the service port on the node.
